swimlane/core/resources/app.py

```diff
@@ -33,8 +33,22 @@ def __init__(self, swimlane, raw):
         self.id = self._raw['id']
         self.tracking_id = self._raw.get('trackingFieldId')
 
-        self._fields_by_id = {f['id']: f for f in self._raw['fields']}
-        self._fields_by_name = {f['name']: f for f in self._raw['fields']}
+        self._fields_by_id = dict()
+        self._fields_by_name = dict()
+        self._defaults = dict()
```

Comment:
The underscore means that it is private member, yet you are using it in record module below. That looks very strange and confusing. I suggest changing to no underscore.

Reply:
I am marking defaults as a private member so that it is more clear that this attribute is not to be used directly in an integration.  In the record_factory method, there are other references to private attributes used as well.

Reply:
Private to record where they are used, not private to other class - app.